lib/shopify_api/auth/session.rb

```diff
@@ -40,6 +46,11 @@ def expired?
         @expires ? @expires < Time.now : false
       end
 
+      sig { returns(T::Boolean) }
+      def refresh_token_expired?
+        @refresh_token_expires ? @refresh_token_expires < Time.now : false
```

Comment:
Should we add a buffer to this?

To avoid race conditions where this returns true but then the exchange fails?